### PR TITLE
Update validateDocument function to validate the entire document and return all validation errors at once

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose

Fixes: [#2848](https://github.com/ballerina-platform/ballerina-standard-library/issues/2848)

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~~Updated the spec~~

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ### Changed
 - [[#3062] Improve Compilation Error Messages to be more specific](https://github.com/ballerina-platform/ballerina-standard-library/issues/3062)
+- [[#2848] Only one kind of errors return from a invalid graphql request](https://github.com/ballerina-platform/ballerina-standard-library/issues/2848)
```
Comment (ThisaruGuruge) [inline changelog.md:23]: Let's use the title case here.

Comment (MohamedSabthar) [reply changelog.md]: updated

File: changelog.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ### Changed
 - [[#3062] Improve Compilation Error Messages to be more specific](https://github.com/ballerina-platform/ballerina-standard-library/issues/3062)
+- [[#2848] Only One Kind of Errors Return From a Invalid GraphQL Request](https://github.com/ballerina-platform/ballerina-standard-library/issues/2848)
```
Comment (ThisaruGuruge) [inline changelog.md:23]:
```suggestion
- [[#2848] Only One Kind of Errors Return From an Invalid GraphQL Request](https://github.com/ballerina-platform/ballerina-standard-library/issues/2848)
```

Comment (ThisaruGuruge) [reply changelog.md:23]: Shall we update the previous entry as well? Seems to have mixed up the title case.

Comment (MohamedSabthar) [reply changelog.md]: Updated with AP style

File: changelog.md [hunk 3]
```diff
@@ -14,12 +14,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - [[#3233] Add Service Level Interceptor Support for GraphQL Subscriptions](https://github.com/ballerina-platform/ballerina-standard-library/issues/3233)
 
 ### Fixed
-- [[#3294] Fix GraphQL dynamic listener is not working for module level service declaration](https://github.com/ballerina-platform/ballerina-standard-library/issues/3294)
-- [[#3375] Fix Multiple Subscription Endpoints with the Same GraphQL Listener is not Working](https://github.com/ballerina-platform/ballerina-standard-library/issues/3375)
+- [[#3294] Fix GraphQL Dynamic Listener Is Not Working for Module Level Service Declaration](https://github.com/ballerina-platform/ballerina-standard-library/issues/3294)
+- [[#3375] Fix Multiple Subscription Endpoints With the Same GraphQL Listener Is Not Working](https://github.com/ballerina-platform/ballerina-standard-library/issues/3375)
```
Comment (ThisaruGuruge) [inline changelog.md:18]: Prepositions should not be capitalized
```suggestion
- [[#3375] Fix Multiple Subscription Endpoints with the Same GraphQL Listener is Not Working](https://github.com/ballerina-platform/ballerina-standard-library/issues/3375)
```

Comment (MohamedSabthar) [reply changelog.md:18]: For AP/APA style title case  `With` and `Is` are capitalized.
Are we using a specific format here https://titlecaseconverter.com/ ?

Comment (ThisaruGuruge) [reply changelog.md:18]: For the title case, the Ballerina team practice is not to capitalize the prepositions AFAIK.

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**